xarray/core/duck_array_ops.py

```diff
@@ -214,26 +214,32 @@ def astype(data, dtype, **kwargs):
 
 
 def asarray(data, xp=np):
+    print(data)
+    print(type(data))
     return data if is_duck_array(data) else xp.asarray(data)
 
 
-def as_shared_dtype(scalars_or_arrays, xp=np):
-    """Cast a arrays to a shared dtype using xarray's type promotion rules."""
-    array_type_cupy = array_type("cupy")
-    if array_type_cupy and any(
-        isinstance(x, array_type_cupy) for x in scalars_or_arrays
-    ):
-        import cupy as cp
-
-        arrays = [asarray(x, xp=cp) for x in scalars_or_arrays]
+def as_duck_array(data, xp=np):
```

Comment:
Is there another function like this anywhere?

Reply:
Yes lets use https://github.com/pydata/xarray/blob/c9ba2be2690564594a89eb93fb5d5c4ae7a9253c/xarray/core/pycompat.py#L132